src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -789,6 +789,11 @@ func (c controller) waitingForClusterVersion() error {
 		operatorStatus, operatorMessage := utils.ClusterOperatorConditionsToMonitoredOperatorStatus(co.Status.Conditions)
 
 		if cvoStatusInService.Status != operatorStatus || (cvoStatusInService.StatusInfo != operatorMessage && operatorMessage != "") {
+			if !timer.Stop() {
```

Comment:
here also you should check for cancellation

Reply:
Why?